### PR TITLE
feat: moved users and auth away from the Store

### PR DESCRIPTION
**Pull request by thiagodallacqua-hpe**
## Description

Tickets:
- WEB-583
- WEB-590

We're migrating away from our current `Store` to a new architecture. In this PR we're migrating `users` and `auth`.

## Test Plan

- Check initial login and routing (which is based on both states);
- Check login and logout flows
- Check user profile interactions

## Checklist
- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: webui/react/src/App.tsx
Comment (ClaireNeveu) [inline webui/react/src/App.tsx:37]: Instead of propagating this fake data, it would be better to check the actual ground truth. Normally I would recommend carrying the loadable out to the use-cases so they can each handle the loading case in the appropriate manner but here you can just make this as simple as
```typescript
const isAuthenticated = Loadable.match(useAuth(), {
  Loaded: (auth) => auth.isAuthenticated,
  NotLoaded: () => false
});
```

Comment (ClaireNeveu) [reply webui/react/src/App.tsx:37]: There are some other places you're still using this "initial state" pattern that I recommend fixing too. `checked` should be replaceable by `Loadable.isLoaded(useAuth())`.

File: webui/react/src/stores/users.tsx
Comment (ClaireNeveu) [inline webui/react/src/stores/users.tsx:27]: Can we split the data and the functions into separate hooks? `useAuth`, `useUsers`, `useResetAuth`, etc? Once we aren't using react context we can make the actions into non-hooks.

File: webui/react/src/stores/users.tsx
Comment (ClaireNeveu) [inline webui/react/src/stores/users.tsx:156]: Why doesn't this get its own file?

**Review by ClaireNeveu — CHANGES_REQUESTED** [review]

File: webui/react/src/hooks/useSettingsProvider.tsx
Comment (ClaireNeveu) [inline webui/react/src/hooks/useSettingsProvider.tsx:37]: This is exactly what I mean about the initState being confusing. We aren't using `isAuthenticated` in this file and the spinner should just check whether the Loadable is Loaded or not. The only value we actually want here is `user`.

This should be `const user = Loadable.map(useAuth().auth, auth => auth.user)` and then we can use that value in the component.

File: webui/react/src/pages/Cluster/ClusterHistoricalUsage.tsx
Comment (ClaireNeveu) [inline webui/react/src/pages/Cluster/ClusterHistoricalUsage.tsx:30]: Another place where we should have a proper loading state.

File: webui/react/src/stores/auth.tsx
Comment (ClaireNeveu) [inline webui/react/src/stores/auth.tsx:64]: This makes me think that `currentUser` should just be it's own bit of state since it is apparently updated independently of the rest of this object.

**Review by ClaireNeveu — CHANGES_REQUESTED** [review]

File: webui/react/src/hooks/useSettingsProvider.tsx
Comment (ClaireNeveu) [reply webui/react/src/hooks/useSettingsProvider.tsx:37]: This isn't a standard, it's just what certain parts of the code looks like. The information we want here is different than in other parts of the PR and so the code should look different.

**Review by EmilyBonar — APPROVED** [review]
LGTM, but you might want to wait for approval from Claire to merge

**Pull request merged**
